### ISSUE TITLE
`single_match` wants to delete my comments

### ISSUE THREAD

**Issue by kpreid** [rust-lang/rust-clippy#8634]
### Summary

I just updated to the Rust 1.60.0 prerelease and got a new lint; when I investigated further I realized that I don't agree with it in other cases that came up previously.

The `single_match` lint detects when a `match` has only one nonempty arm, and proposes replacing it with an `if let` with no `else`. However, the suggestion deletes any comments present in the empty (of code) arm; such comments are very useful for documenting “why there is nothing needing doing”.

Additionally, the pattern of the second arm may be useful compiler-checked-documentation; for example, the pattern `Err(())` ensures that the code must be changed if the error type stops being `()`, and has a different connotation than `Err(_)` (which a reader might infer after seeing an `if let Ok(_) = ... {...} else {...}`).

Proposal: The `single_match` lint should not trigger if there are any comments in the "empty" arms.

### Lint Name

single_match

### Reproducer

Code:

```rust
struct SomeError(i32, i32);

fn foo(x: Result<i32, ()>) {
    match x {
        Ok(y) => {
            println!("Yay! {y}");
        }
        Err(()) => {
            // Ignore this error because blah blah blah.
        }
    }
}

fn bar(x: Result<i32, SomeError>) {
    match x {
        Ok(y) => {
            println!("Yay! {y}");
        }
        Err(_) => {
            // TODO: Process the error properly.
        }
    }
}
```

Output:

```
warning: you seem to be trying to use `match` for destructuring a single pattern. Consider using `if let`
  --> src/lib.rs:4:5
   |
4  | /     match x {
5  | |         Ok(y) => {
6  | |             println!("Yay! {y}");
7  | |         }
...  |
10 | |         }
11 | |     }
   | |_____^
   |
   = note: `#[warn(clippy::single_match)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_match
help: try this
   |
4  ~     if let Ok(y) = x {
5  +         println!("Yay! {y}");
6  +     }
   |

warning: you seem to be trying to use `match` for destructuring a single pattern. Consider using `if let`
  --> src/lib.rs:15:5
   |
15 | /     match x {
16 | |         Ok(y) => {
17 | |             println!("Yay! {y}");
18 | |         }
...  |
21 | |         }
22 | |     }
   | |_____^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_match
help: try this
   |
15 ~     if let Ok(y) = x {
16 +         println!("Yay! {y}");
17 +     }
   |
```

Note that both of the suggestions discard the comments.

### Version

```text
rustc 1.60.0 (7737e0b5c 2022-04-04)
binary: rustc
commit-hash: 7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c
commit-date: 2022-04-04
host: x86_64-apple-darwin
release: 1.60.0
LLVM version: 14.0.0

Also tested with rustc 1.59.0 on the Playground (above output is from there)
```

**Comment by kpreid** [comment]
(I'm not sure if this counts as a bug, exactly, but the "false positive" template seemed most appropriate for what I wanted to say.)
